cmd/cmd.go

```diff
@@ -169,6 +169,12 @@ var globalFlags = []cli.Flag{
 		Value:  "",
 		EnvVar: "GDRIVE_CLIENT_JSON_FILEPATH",
 	},
+	cli.StringFlag{
+		Name:   "gdrive-auth-type",
```

Comment:
this way you pass `gdrive-client-json-filepath` when it's a `gdrive-service-account-filepath` indeed

also it's a breaking change, I cannot accept it, sorry :(

you can add `gdrive-service-account-filepath`, check that only this or `gdrive-client-json-filepath` is set (and at least one of them), and set the `authType` value to pass to the factory: https://github.com/dutchcoders/transfer.sh/pull/512/files#diff-8e494a434a8037b6c0b888e25b2baae7618fe65e792d4a155dadd096e9350667R488

please, remember to change the name of the param in the factory to something generic, like `gdriveCredentialsFilepath`